parse_ui.py

```diff
@@ -0,0 +1,79 @@
+import os
+import re
+import xml.etree.ElementTree as ET
+
+import inspect
+import importlib
+
+# utility script for parsing .ui files for type information
+
+# collect ui files
+ui_files = [file for file in os.listdir() if file.endswith('.ui')]
+
+# collect .py files
+py_files = [file for file in os.listdir() if file.endswith('.py')]
+
+# pair ui with py files (ui CamelCase to py snake_case)
+pairs = []
+for ui_file in ui_files:
+    for py_file in py_files:
+        if py_file.split('.')[0] == re.sub(r'(?<!^)(?=[A-Z])', '_', ui_file.split('.')[0]).lower():
+            pairs.append((ui_file, py_file))
+            break
```

Comment:
That's painful. I feel like it should be possible to just iterate over each file in the directory and check if it's a .py file and if it matches a camelCased .ui file inside ui_files (or the other way around). Doubt there would be much of a speed improvement here but it might look cleaner.

Reply:
yeah I think there's definitely a better way of doing this, but I'll probably just leave `parse_ui.py` the mess that it is for now.  It's basically just for development since it adds type hints from ui files into classes so I know widget names, get type-based function info, etc.  It's existence or cleanliness doesn't affect AMP itself, just the dev process.

That being said, if there end up being way more ui files and way more amp plugins, I'll probably clean it up a bit cause it is actually kinda slow (I think the mega-mondo slow part is actually the inspect/importlib loop tho)